packages/core/strapi/lib/Strapi.js

```diff
@@ -32,39 +43,53 @@ const LIFECYCLES = {
 
 class Strapi {
   constructor(opts = {}) {
-    this.container = createContainer(this);
-
     this.dir = opts.dir || process.cwd();
-    this.config = loadConfiguration(this.dir, opts);
+    const appConfig = loadConfiguration(this.dir, opts);
+    this.container = createContainer(this);
+    this.container.register('config', createConfigProvider(appConfig));
+    this.container.register('content-types', contentTypesRegistry(this));
+    this.container.register('services', servicesRegistry(this));
+    this.container.register('policies', policiesRegistry(this));
+    this.container.register('middlewares', middlewaresRegistry(this));
+    this.container.register('controllers', controllersRegistry(this));
+    this.container.register('modules', modulesRegistry(this));
 
+    this.isLoaded = false;
     this.reload = this.reload();
-
-    // Expose `koa`.
     this.app = new Koa();
     this.router = new Router();
-
     this.server = createHTTPServer(this, this.app);
-
-    this.app.proxy = this.config.get('server.proxy');
-
-    // Logger.
-    const loggerUserConfiguration = this.config.get('logger', {});
-    this.log = createLogger(loggerUserConfiguration);
-
-    this.isLoaded = false;
-
-    // internal services.
+    this.plugins = {}; // to remove V3
+    this.contentTypes = {}; // to remove V3
     this.fs = createStrapiFs(this);
     this.eventHub = createEventHub();
     this.startupLogger = createStartupLogger(this);
+    this.app.proxy = this.config.get('server.proxy');
+    this.log = createLogger(this.config.get('logger', {}));
 
     createUpdateNotifier(this).notify();
   }
 
+  get config() {
+    return this.container.get('config');
+  }
+
   get EE() {
     return ee({ dir: this.dir, logger: this.log });
   }
 
+  service(uid) {
+    return this.container.get('services').get(uid);
+  }
+
+  plugin(name) {
+    return this.container.get('modules').get(`plugin::${name}`);
+  }
+
+  get pluginsV4() {
```

Comment:
Can you document what is left to do to be able to just have plugins here

Reply:
Still needed?

Reply:
Nope :)